### ISSUE TITLE
Customize the logo / Change the name of Overseerr

### ISSUE THREAD

**Issue by GhostPyxel** [sct/overseerr#404]
**Is your feature request related to a problem? Please describe.**
One of the simpliest way to customize Ombi as a User nowadays is to replace the Logo and Name of the application (showing up in the tab next to the favicon). I would be nice if the admin could do that in Overseerr as well.

**Describe the solution you'd like**
Within the "General Settings" Section, create a subsection named "Customization" (or else) to upload a .png or .jpg for the new logo that will overwrite the existing one. Same thing could be done for a text input where you could write a new name for the application and rename Overseerr.

**Comment by ctrlaltd1337ed** [comment]
+1 to this, changing the icon and name of the app would be great.

**Comment by Cyb3rGh05t** [comment]
+1 for this one

**Comment by Albeyon** [comment]
+1 for this! Maybe the option to change the SSL too?

**Comment by morganzero** [comment]
+1 for this!

**Comment by WoisWoi** [comment]
+1
Indeed, it would be really nice to have this :)

**Comment by Cajs** [comment]
Just to feedback on this, I can see the ability to change the name of Overseer in the General Settings via "Application Title", but cannot see the upload replacement. Is this still outstanding?

**Comment by Emalton** [comment]
Where can the logo be changed?

**Comment by PhAzE-Variance** [comment]
If the ability to add a custom logo to the top right of Overseerr get's implemented, can we also have the option to use the logo in the body of email notifications, instead of the application title?

**Comment by TheCatLady** [comment]
> If the ability to add a custom logo to the top right of Overseerr get's implemented, can we also have the option to use the logo in the body of email notifications, instead of the application title?

Please follow #405 for custom notification support.

**Comment by Dabu-dot** [comment]
No

**Comment by carnivorouz** [comment]
Agreed on this.  If I do transition from Ombi it will be with just about 50 users and something that now says "Overseerr" instead will traumatize them unfortunately.

**Comment by dduives** [comment]
Hi, I would really love this feature still. My users are getting confused by the completely different logo/name used for my services.
Thanks!

**Comment by KJThaDon** [comment]
Easiest code to write and still not a thing. Wonder why

**Comment by sct** [comment]
> Easiest code to write and still not a thing. Wonder why

Feel free to PR it! Contributions are always welcome.

**Comment by Dabu-dot** [comment]
Go home stalebot, you're drunk.

**Comment by Emalton** [comment]
Ver helpful, thanks!!

**Comment by senorbueno** [comment]
The most appropriate path right now is to create a custom logos into a local public/ directory and bind mount that folder to the created docker instance like @meleniumshane90 mentioned. I just append this line to my docker startup/update script:

`-v ~/.overseerr/public:/app/public \`

In there I have my custom SVGs and other image files. The problem with updating SVGs on-the-fly for overseerr is that they don't use embedded (rasterized) images like .png/.jpg for the logos, they use an XML-based text format to describe how the image should appear. This allows for browser compatibility and scaling/resizing, since they're resolution independent. It takes a little Adobe Illustrator work but is totally doable and my login/home screen logos look awesome.

**Comment by mastertonux** [comment]
Jealous of you Docker users. Us Snap folks can't do this. Really hoping to see this feature added at some point in the UI because my users still get confused about what Overseerr is when the arrive at the site.

**Comment by rdeutsch03** [comment]
> The most appropriate path right now is to create a custom logos into a local public/ directory and bind mount that folder to the created docker instance like @meleniumshane90 mentioned. I just append this line to my docker startup/update script:
> 
> `-v ~/.overseerr/public:/app/public \`
> 
> In there I have my custom SVGs and other image files. The problem with updating SVGs on-the-fly for overseerr is that they don't use embedded (rasterized) images like .png/.jpg for the logos, they use an XML-based text format to describe how the image should appear. This allows for browser compatibility and scaling/resizing, since they're resolution independent. It takes a little Adobe Illustrator work but is totally doable and my login/home screen logos look awesome.

hey @senorbueno I appreciate your help with this work around! I was able to get the login screen logo (logo_stacked.svg) and the favicons (favicon.ico, favicon-32x32.png, and favicon-16x16.png) successfully replaced with custom logos, but the logo in the top left corner once logged in (logo_full.svg) is rendering as 0 x 0 pixels and is thus not showing up. is there another change I need to make with the .svg file or the XML somewhere to make this logo render properly?

**Comment by rdeutsch03** [comment]
In case someone needed an answer to my question, I figured it out. I had to add width and height parameters to the code of the .svg file and it rendered properly.

**Comment by CForChrisProooo** [comment]
If this gets added, I donate.

**Comment by dduives** [comment]
I agree, if this get added I will also donate

**Comment by Emalton** [comment]
I agree too and would also donate.

**Comment by tscibilia** [comment]
I would also like to mention that the "custom application title" update from v.1.19 doesn't change the Overseerr name displayed when a user attempts to login via the plex auth screen. I believe the expected behavior would also change the application title name on the login screen. 
![image](https://user-images.githubusercontent.com/585808/181844552-0c1c55a8-d597-4546-92ec-1132441f2d75.png)

**Comment by Emalton** [comment]
Definitely don't close this!

**Comment by cavfeet** [comment]
> Container Path: /app/public/logo_full.svg -> Host Path: /path/to/custom/logo.sv

what do you mean with In docker, add patch. Could you explain in more datail trying to use you manuel fix.

**Comment by WoisWoi** [comment]
> > Container Path: /app/public/logo_full.svg -> Host Path: /path/to/custom/logo.sv
> 
> what do you mean with In docker, add patch. Could you explain in more datail trying to use you manuel fix.

That's what he meant (example from Portainer), you have to remap the volume via docker : 
![image](https://user-images.githubusercontent.com/75491231/192720625-0e7d3b48-cc11-4c81-a81e-808d18d8ee10.png)

**Comment by Jmcguire525** [comment]
+1 for keeping this open and making it customizable

**Comment by collinsimler** [comment]
+1 still want this in 2023 please

**Comment by xunil75** [comment]
+1

**Comment by luizfeliperc** [comment]
> In case anyone wants to do this manually, this is a temporary solution: In Docker, add a path: `Container Path: /app/public/logo_full.svg -> Host Path: /path/to/custom/logo.svg`
> 
> If something changes, this is the most current file structure: https://github.com/sct/overseerr/tree/develop/public

I used it this way for more than 1 year and today I went to change the logo for a more current one and I can't do it anymore. Even recreating the container it pulls the old/original image. I do not know what else to do.

**Comment by SpoooodaLs** [comment]
> > In case anyone wants to do this manually, this is a temporary solution: In Docker, add a path: `Container Path: /app/public/logo_full.svg -> Host Path: /path/to/custom/logo.svg`
> > If something changes, this is the most current file structure: https://github.com/sct/overseerr/tree/develop/public
> 
> I used it this way for more than 1 year and today I went to change the logo for a more current one and I can't do it anymore. Even recreating the container it pulls the old/original image. I do not know what else to do.

I use this way currently but man it was a pain in the neck to get the SVG file to work properly, I can't remember the format but it had to match some specific type of format that the original had.

I don't have much knowledge of SVG files, so it took me a few hours to get it working.

Are you using Unraid? If so, I replace `/app/overseerr/public/logo_full.svg`

**Comment by luizfeliperc** [comment]
> 

No, I'm using Linux's standard docker and docker-compose.
I don't know if it happens to anyone else, but here I am commenting on the volumes and leaving the original images, they are not reflected in the system, even clearing the browser's cache I still see the edited images

**Comment by luizfeliperc** [comment]
> > > In case anyone wants to do this manually, this is a temporary solution: In Docker, add a path: `Container Path: /app/public/logo_full.svg -> Host Path: /path/to/custom/logo.svg`
> > > If something changes, this is the most current file structure: https://github.com/sct/overseerr/tree/develop/public
> > 
> > 
> > I used it this way for more than 1 year and today I went to change the logo for a more current one and I can't do it anymore. Even recreating the container it pulls the old/original image. I do not know what else to do.
> 
> I use this way currently but man it was a pain in the neck to get the SVG file to work properly, I can't remember the format but it had to match some specific type of format that the original had.
> 
> I don't have much knowledge of SVG files, so it took me a few hours to get it working.
> 
> Are you using Unraid? If so, I replace `/app/overseerr/public/logo_full.svg`

No, I'm using Linux's standard docker and docker-compose.
I don't know if it happens to anyone else, but here I am commenting on the volumes and leaving the original images, they are not reflected in the system, even clearing the browser's cache I still see the edited images

**Comment by misterduve** [comment]
I could never get it to work. Docker/Portainer/Overseer

**Comment by tungmeister** [comment]
@wnconnor the path has changed, it's now /app/overseerr/public/logo_full.svg

**Comment by cavfeet** [comment]
> In case anyone wants to do this manually, this is a temporary solution: In Docker, add a path: `Container Path: /app/public/logo_full.svg -> Host Path: /path/to/custom/logo.svg`
> 
> If something changes, this is the most current file structure: https://github.com/sct/overseerr/tree/develop/public

How do you add an container path ?
I am use a docker compose that looks like this:

  overseerr:
      <<: *common-keys-media # See EXTENSION FIELDS at the top
      image: lscr.io/linuxserver/overseerr:latest
      container_name: overseerr
      ports:
        - "5055:5055"
      volumes:
        - $DOCKERDIR/appdata/overseerr:/config
      environment:
        <<: *default-tz-puid-pgid

**Comment by nielsadolf** [comment]
Will donate if this is added, will use ombi until this is added tbh, looks sooo bad being stuck with the Overseerr logo and name at the top...

**Comment by KJThaDon** [comment]
This is how I do it in a docker on unraid. Be aware they keep marking comments as spam on here so you don't find this out.

![Screenshot_2](https://user-images.githubusercontent.com/1257000/221443236-a3ddace2-47ce-42ae-9cf7-ee761fe60dca.png)

**Comment by SpartacusIam** [comment]
> This is how I do it in a docker on unraid. Be aware they keep marking comments as spam on here so you don't find this out.
> 
> ![Screenshot_2](https://user-images.githubusercontent.com/1257000/221443236-a3ddace2-47ce-42ae-9cf7-ee761fe60dca.png)

I wish the devs would just say they are against this because they want to advertise their app.  Them they can just ose this request as opposed to pretending like it s a possible feature coming and secretly marking comments as spam.

**Comment by KJThaDon** [comment]
> > This is how I do it in a docker on unraid. Be aware they keep marking comments as spam on here so you don't find this out.
> > ![Screenshot_2](https://user-images.githubusercontent.com/1257000/221443236-a3ddace2-47ce-42ae-9cf7-ee761fe60dca.png)
> 
> I wish the devs would just say they are against this because they want to advertise their app. Them they can just ose this request as opposed to pretending like it s a possible feature coming and secretly marking comments as spam.

Exactly

**Comment by Emalton** [comment]
@TheCatLady Would this issue be suitable for the never-stale label?

**Comment by KJThaDon** [comment]
> I would also like to mention that the "custom application title" update from v.1.19 doesn't change the Overseerr name displayed when a user attempts to login via the plex auth screen. I believe the expected behavior would also change the application title name on the login screen. ![image](https://user-images.githubusercontent.com/585808/181844552-0c1c55a8-d597-4546-92ec-1132441f2d75.png)

Anyone know of a fix for this?

**Comment by carnivorouz** [comment]
> > This is how I do it in a docker on unraid. Be aware they keep marking comments as spam on here so you don't find this out.
> 
> How do you do it in portainer?

If you scroll up to my earlier comment, I put in a imgur link to my portainer setup with this working.

**Comment by remserwis** [comment]
I've got logo_full.svg (for top left corner)  and logo_stacked.svg (for login screen)  working properly with docker folder mapping. Great stuff....Unfortunately emails still coming with original instead of putting logo_full.png the same way on server. Name of the file in email is correct...I am on 1.32.5 version....Do you think it's hard linked now on sending bot ?

**Comment by jakereed1** [comment]
> I've got logo_full.svg (for top left corner) and logo_stacked.svg (for login screen) working properly with docker folder mapping. Great stuff....Unfortunately emails still coming with original instead of putting logo_full.png the same way on server. Name of the file in email is correct...I am on 1.32.5 version....Do you think it's hard linked now on sending bot ?

Hi @remserwis . How did you get it working in docker folder mapping? I've tried mapping a svg file using docker-compose to /app/overseerr/public/logo_full.png and /app/public/logo.png but neither seem to change the default logo.

**Comment by the-robbie** [comment]
> > I've got logo_full.svg (for top left corner) and logo_stacked.svg (for login screen) working properly with docker folder mapping. Great stuff....Unfortunately emails still coming with original instead of putting logo_full.png the same way on server. Name of the file in email is correct...I am on 1.32.5 version....Do you think it's hard linked now on sending bot ?
> 
> 
> 
> Hi @remserwis . How did you get it working in docker folder mapping? I've tried mapping a svg file using docker-compose to /app/overseerr/public/logo_full.png and /app/public/logo.png but neither seem to change the default logo.

Using Docker, I mapped /MY_FILE_DIRECTORY/public to /app/public as well as /MY_FILE_DIRECTORY/public/logo_full.svg to /app/public/logo_full.svg. This is using my custom logo_full.svg for the top left corner, as well as using other my other custom logo files in the public directory such as logo_stacked.svg for the login screen and favicon.ico, favicon-16x16.png, and favicon-32x32.png for my custom favicon. I implemented this about a year ago and it has been working well ever since.

If you did something similar to this with a logo_full.png and it's not working, maybe you need to include a logo_full.svg file as well. I believe the .svg and .png files are used for different use cases, so if you don't have a .svg file you probably need one as well. Otherwise you could try mapping your .png file to the .svg file, but I'm not sure that would work as I haven't tried it myself.

**Comment by jakereed1** [comment]
> 

Thanks for getting back to me. When I wrote .png in my previous reply I did mean .svg. I hadn't tried mapping the folder to public though as well as the file. However I've just tried that but still no joy.. I can live without the custom logo but thanks anyway.
`    volumes:
      - /var/docker/overseerr/config:/config
      - /var/docker/overseerr/config:/app/public
      - /var/docker/overseerr/config/logo_full.svg:/app/public/logo_full.svg`

**Comment by the-robbie** [comment]
I see you mapped all overseer directories to one local config directory. I also noticed you didn't specify overseer's /app directory on your config mapping. For reference, I separated them out into similarly named local directories. This was how I mapped the volumes on my container:

```
volumes:
  - /var/docker/overseerr/config:/app/config
  - /var/docker/overseerr/public:/app/public
  - /var/docker/overseerr/public/logo_full.svg:/app/public/logo_full.svg
  - /var/docker/overseerr/src:/app/src
```

**Comment by remserwis** [comment]
Hi...I wanted to send you a screenshot from portainer, but it looks since I just updated tautulli docker my mappings are gone. I will redo them and paste a picture later.  I remember it didn't want to work with mapping of the folder for me...I mapped file2file for logos.

**Comment by jakereed1** [comment]
cheers guys. I was using the docker-compose example from the overseerr website which only mentions a config folder
https://docs.overseerr.dev/getting-started/installation#:~:text=Docker%20Compose-,For,-details%20on%20how

I've just added the folders / mapping exactly as you did but still no joy. I also tried mapping a svg file directly to /app/public/logo.svg and /app/overseerr/public/logo_full.svg (with only my previous config folder) and the logo then disappears from Overseerr which feels like progress. I've tried multiple svg files of varying resolutions.

I'm on the latest image 1.32.5.

**Comment by remserwis** [comment]
if logo disappeared, it's probably a good sign that you did correct mapping.  now just create proper svg.  From my experience: 
1) I downloaded original logo end edited it to mine (to keep size etc. the same)
2) when exporting there was a setting about logo positioning ( my soft was centering it by default but it should be in top left corner )

**Comment by jakereed1** [comment]
Cheers guys. Seems like the resolution of the svg may be the key. I got it working using 300 x 92 svg and the below container mappings.
    volumes:
      - /var/docker/overseerr/config:/config
      - /var/docker/overseerr/config/logo_full.svg:/app/public/logo_full.svg

I tried to change the logo on the login screen by mapping a svg file the same size as the original stacked svg file (332 x 165) but couldn't get this logo to change. I tried mapping this using the below:
      - /var/docker/overseerr/config/logo_stacked.svg:/app/public/logo_stacked.svg
      
I also tried changing the logo in the email by using a png file as recommended in a post above but couldn't get this to work either.
I think I will call it a day, I got 1 out of 3 changed and spent a lot longer on this than I planned to!

**Comment by Jmcguire525** [comment]
I wish this would get implemented in an easy to use way... I don't know how to code it but if I hired someone to do that and submit a PR would it be accepted?

**Comment by AIndoria** [comment]
What absurdity that most of the comments are marked as spam or offtopic. Devs should be better than this.

**Comment by SpartacusIam** [comment]
> What absurdity that most of the comments are marked as spam or offtopic. Devs should be better than this. People sometimes want to have fun names for their instances.

Because it's very clear that not only do the devs have no interest in implementing this change, they are obviously against it.

**Comment by adamk33n3r** [comment]
> > What absurdity that most of the comments are marked as spam or offtopic. Devs should be better than this. People sometimes want to have fun names for their instances.
> 
> Because it's very clear that not only do the devs have no interest in implementing this change, they are obviously against it.

If they are against it, they should close this issue then.

**Comment by SpartacusIam** [comment]
> > > What absurdity that most of the comments are marked as spam or offtopic. Devs should be better than this. People sometimes want to have fun names for their instances.
> > 
> > 
> > Because it's very clear that not only do the devs have no interest in implementing this change, they are obviously against it.
> 
> If they are against it, they should close this issue then.

Well, if they were okay with it being publicly known that they are against it, then yes this thread would have been closed a long time ago and comments wouldn't be continuously being marked as spam.  Clearly they don't want their users to know that they are against their users using any kind of custom branding.  I guess is they feel like they're branding is what promotes their software. But I really don't see that as being the case.

**Comment by adamk33n3r** [comment]
> > > > What absurdity that most of the comments are marked as spam or offtopic. Devs should be better than this. People sometimes want to have fun names for their instances.
> > > 
> > > 
> > > Because it's very clear that not only do the devs have no interest in implementing this change, they are obviously against it.
> > 
> > 
> > If they are against it, they should close this issue then.
> 
> Well, if they were okay with it being publicly known that they are against it, then yes this thread would have been closed a long time ago and comments wouldn't be continuously being marked as spam. Clearly they don't want their users to know that they are against their users using any kind of custom branding. I guess is they feel like they're branding is what promotes their software. But I really don't see that as being the case.

Ya, it's not like the people that are using my instance of Overseer are EVER going to set up their own 😂 that's why they're using mine! So not sure who they'd be marketing to

**Comment by Staticxtasy** [comment]
Ridiculous that this is still an issue three years later.  My users are very turned off by the overseerr logo plastered all over the app.

**Comment by sct** [comment]
We mark comments as off-topic because they are. Constantly asking for a feature doesn't make us more aware of it. We work in our free time. We are aware of the ticket and want to support it eventually. It's just not the top priority right now.

**Comment by Kevin1Smid** [comment]
For those who want a temporary fix, in NGINX just set:
location /logo_full.svg {
proxy_pass UrlToSvg.svg;
}

**Comment by morganzero** [comment]
**To customize the project logos, follow these steps:**

Locate the default logos in the project code under the following paths:
```
        /public/logo_full.png
        /public/logo_full.svg
        /public/logo_stacked.svg
        /public/os_logo_filled.svg
        /public/overseerr_poster_not_found_logo_top.svg
        /public/overseerr_poster_not_found_logo_center.svg
```
If you're using Docker and building the container, ensure that the public folder is copied to /app/overseerr/public.

To deploy the project using docker-compose, bind your custom logo files to the application by adding this line to your docker-compose.yml file:
```yaml
volumes:
  - "/path/to/your/custom/logos/:/app/overseerr/public"
```

The docker-compose example would look like this instead:
```yaml
version: '3'
services:
  overseerr:
    build:
      context: .
      dockerfile: Dockerfile.local
    ports:
      - 5055:5055
    volumes:
      - "/path/to/your/custom/logos/:/app/overseerr/public"
      - .:/app:rw,cached
      - /app/node_modules
      - /app/.next
```

Rename your custom logos with the following exact names to replace the defaults:
```
        logo_full.png
        logo_full.svg
        logo_stacked.svg
        os_logo_filled.svg
        overseerr_poster_not_found_logo_top.svg
        overseerr_poster_not_found_logo_center.svg
```
By following these steps, you can easily customize the logos for the project.

**TL;DR:**
To deploy using docker-compose, follow the example above, and ensure your custom logos are placed in the correct path within the project.

![Alt Text](https://media.tenor.com/zX-E5TKYEikAAAAM/three-finger-salute-good-luck.gif)

**Comment by kalyway101** [comment]
For anyone who is finding some conflicting information in previous versions/comments... the current location that logos are pulled from is `/app/public/`. It no longer is pulling from `/app/overseerr/public` as of version **1.33.2**.

**Comment by jordanjones98** [comment]
> > > I would also like to mention that the "custom application title" update from v.1.19 doesn't change the Overseerr name displayed when a user attempts to login via the plex auth screen. I believe the expected behavior would also change the application title name on the login screen. ![image](https://user-images.githubusercontent.com/585808/181844552-0c1c55a8-d597-4546-92ec-1132441f2d75.png)
> > 
> > 
> > Anyone know of a fix for this?
> 
> have you know how change it ?

You can change it on line 62 of `src/utils/plex.ts` (at the time of writing this) I looked into making it the same as the `Application Title` but that is a little challenging because that is a DB setting which you need to be authed to see, and of course you aren't authed when you are trying to login. There are a few workarounds I have thought about, but not sure exactly what I will try to implement yet.

**Comment by jordanjones98** [comment]
I have a branch on my fork that allows you to change the text but not the logo. I still need to clean up the CSS to make it work better with longer names, but I will PR it soon.

**Comment by almulder** [comment]
> This is how I do it in a docker on unraid.
> 
> **Be aware they keep marking comments as spam on here so you don't find this out.**
> 
> ![Screenshot_2](https://user-images.githubusercontent.com/1257000/221443236-a3ddace2-47ce-42ae-9cf7-ee761fe60dca.png)

Does this still work for you? I have tried with the new location path, but it does not seem to work.

![image](https://github.com/sct/overseerr/assets/5806190/18843e2b-8c55-4beb-b16b-1437f52b119e)

**Comment by almulder** [comment]
> I have a branch on my fork that allows you to change the text but not the logo. I still need to clean up the CSS to make it work better with longer names, but I will PR it soon.
> 
> I do have a PR open for the name of the application on the plex login page #3811

Man this will be nice to have done, (Also hopefully changes the name in emails to "Open Server Name")

For images I did what a user did above (for Unraid).

And I Would love to test the name change

**Comment by ldejavul** [comment]
> I've got logo_full.svg (for top left corner) and logo_stacked.svg (for login screen) working properly with docker folder mapping. Great stuff....Unfortunately emails still coming with original instead of putting logo_full.png the same way on server. Name of the file in email is correct...I am on 1.32.5 version....Do you think it's hard linked now on sending bot ?

In case anyone finds this troubleshooting your email logo change. Clear the cache for your domain. I spent a stupid  amount of time wondering why my full_logo.png wasn't changing in emails.

**Comment by lnhrdt** [comment]
Hey bot, this is not stale! There's recent work being done in #3811 by @jordanjones98 and @onedr0p. According to the automated checks in that PR, it's awaiting code owner review from @OwsleyJr, @TheCatLady, @danshilm, and/or @sct.

**Comment by lnhrdt** [comment]
Noticed the new release last month, thanks for that @sct! With the project seeing some fresh momentum, any chance we could get a code owner review on #3811?

**Comment by lnhrdt** [comment]
@jordanjones98 closed his own pull request in February but out of respect to his contribution and because of all of the interest in this enhancement seen here in this issue, any chance of getting a maintainer's eyes on #3811?

I wouldn't nudge so much except for the fact that it seemed that the PR was only awaiting code owner review from @OwsleyJr, @TheCatLady, @danshilm, and/or @sct.

**Comment by Damienlee69** [comment]
This feature and music are holding me back from switching from Ombi.
